src/urllib3/connectionpool.py

```diff
@@ -757,18 +773,30 @@ def urlopen(  # type: ignore[override]
             BaseSSLError,
             SSLError,
             CertificateError,
-            HTTPSProxyError,
+            ProxyError,
         ) as e:
             # Discard the connection for these exceptions. It will be
             # replaced during the next _get_conn() call.
             clean_exit = False
             new_e: Exception = e
             if isinstance(e, (BaseSSLError, CertificateError)):
                 new_e = SSLError(e)
-            elif isinstance(e, (OSError, NewConnectionError)) and self.proxy:
-                new_e = ProxyError("Cannot connect to proxy.", e)
-            elif isinstance(e, (OSError, HTTPException)):
-                new_e = ProtocolError("Connection aborted.", e)
+            if (
+                isinstance(
+                    new_e,
+                    (
+                        OSError,
+                        NewConnectionError,
+                        TimeoutError,
+                        SSLError,
+                        HTTPException,
```

Comment:
Not sure if `HTTPException` should be here.

Reply:
I think it makes sense. We can generate an HTTP exception when using the proxy so it makes sense to wrap it too.